lib/js/components/ProgressDonutChart/ProgressDonutChart.vue

```diff
@@ -0,0 +1,285 @@
+		stroke-width: $progress-donut-chart-circle-stroke-width;
+		// Disabled for v3.0
+		//stroke-linecap: round;
+		//stroke-linejoin: round;
+		transition: all $default-transition-time ease-out;
+		transform: rotate(90deg);
+		transform-origin: 50% 50%;
+	}
+
+	&__thumb {
+		stroke: $color-neutral-data-ghost;
+	}
+
+	&__track {
+		@for $i from 1 through 100 {
+			&[data-value='#{$i}'] {
+				stroke-dashoffset: #{$progress-donut-chart-circle-circumference -
+					($progress-donut-chart-circle-circumference * ($i / 100))};
+			}
+		}
+
+		@each $class, $color-name in $progress-donut-chart-range-colors {
+			&.-#{$class} {
+				stroke: $color-name;
+			}
+		}
+
+		stroke-dasharray: $progress-donut-chart-circle-circumference;
+	}
+
+	&__loader {
+		stroke-dasharray: $progress-donut-chart-circle-circumference;
+		stroke: $color-neutral-background-strong;
+		stroke-dashoffset: #{$progress-donut-chart-circle-circumference -
+			($progress-donut-chart-circle-circumference * (25 / 100))};
+		animation: a-spinAround 2s infinite linear;
+		transform: rotate(0deg);
+	}
+
+	&__label {
+		@include info-m-default-bold;
+
+		display: flex;
+		position: absolute;
+		top: 0;
+		left: 0;
+		width: 100%;
+		height: 100%;
+		justify-content: center;
+		flex-wrap: wrap;
+		align-content: center;
+		color: $color-neutral-text;
+	}
+
+	&__icon {
+		color: $color-success-text;
+	}
+
+	&__labelText {
+		&.-success {
+			color: $color-success-text;
+		}
+	}
+
+	&__loaderText {
+		position: relative;
+		width: 2px;
+		height: 2px;
+		margin-top: $space-xxxs;
+		border-radius: 100%;
+		background-color: $color-neutral-text;
+		color: $color-neutral-text;
+		animation: dot-flashing 1s infinite linear alternate;
+		animation-delay: 0.5s;
+
+		&::before,
+		&::after {
+			content: '';
+			display: inline-block;
+			position: absolute;
+			top: 0;
+		}
+
+		&::before {
+			left: -4px;
+			width: 2px;
+			height: 2px;
+			border-radius: 100%;
+			background-color: $color-neutral-text;
+			color: $color-neutral-text;
+			animation: dot-flashing 1s infinite alternate;
+			animation-delay: 0s;
+		}
+
+		&::after {
+			left: 4px;
+			width: 2px;
+			height: 2px;
+			border-radius: 100%;
+			background-color: $color-neutral-text;
+			color: $color-neutral-text;
+			animation: dot-flashing 1s infinite alternate;
+			animation-delay: 1s;
+		}
+	}
+}
+</style>
+
+<script lang="ts">
+import { PROGRESS_DONUT_CHART_STATES, ProgressDonutChartRange } from './ProgressDonutChart.consts';
+import { PropType } from 'vue';
+import DsIcon, { ICON_SIZES, ICONS } from '../Icons/Icon';
+
+export default {
+	name: 'ProgressDonutChart',
+	components: {
+		DsIcon,
+	},
+	props: {
+		label: { type: String, default: '' },
+		state: {
+			type: String,
+			default: PROGRESS_DONUT_CHART_STATES.DEFAULT,
+			validator(state) {
+				return Object.values(PROGRESS_DONUT_CHART_STATES).includes(state);
+			},
+		},
+		isDone: { type: Boolean, default: false },
+		hasOverage: { type: Boolean, default: false },
```

Comment:
@rogatty suggested to remove these props and use only `state`. Did you discuss it with UX?